moveit_ros/planning/moveit_cpp/include/moveit/moveit_cpp/planning_component.h

```diff
@@ -73,6 +75,33 @@ class PlanningComponent
     }
   };
 
+  class PlanSolutions
+  {
+  public:
+    /// \brief Constructor
+    PlanSolutions(const size_t expected_size = 0)
+    {
+      solutions_.reserve(expected_size);
+    }
+
+    /// \brief Thread safe method to add PlanSolutions to this data structure
+    void pushBack(PlanSolution plan_solution)
```

Comment:
You may want to change this later to `insert`, where the `solutions_` container is kept in sorted order according to some solution ranking mechanism. See [OMPL](https://github.com/ompl/ompl/blob/main/src/ompl/base/src/ProblemDefinition.cpp#L54-L161) for an example. It be neat if the ranking can be user-specified. Maybe that's overkill and maybe the logic for extracting the best solution can be combined with the logic for terminating planning when a "good enough" solution has been found.

Reply:
That's a good idea but I feel like this should not block this PR. I've added a TODO for now